crates/nu-explore/src/views/record/mod.rs

```diff
@@ -332,30 +336,36 @@ enum UIMode {
 }
 
 #[derive(Debug, Clone)]
-pub struct RecordLayer<'a> {
-    columns: Cow<'a, [String]>,
-    records: Cow<'a, [Vec<Value>]>,
+pub struct RecordLayer {
+    column_names: Vec<String>,
+    // the sole reason we keep values so we could return original value.
+    // if it's being peaked.
+    //
+    // if not we could accept an iterator over it.
+    // or if it could be Clonable we could do that anyway;
+    // cause it would keep memory footprint lower while keep everything working
+    // (yee would make return O(n); we would need to traverse iterator once again; but maybe worth it)
+    record_values: Vec<Vec<Value>>,
+    // it's an Optional so we could not have any configurations passed at creation stage.
```

Comment:
I've run into this problem before. It would be nice to pass config into view+layer constructors instead of having to pass config later.

Can I make a PR for that another time? Are you aware of any reason why that would not work?

Reply:
Answering my own question: I'm guessing the "can't pass config in at creation time" limitation comes from when `explore` config could be changed on the fly. So I'm probably OK to change that behaviour.